app/controllers/application_controller.rb

```diff
@@ -21,6 +23,23 @@ def configure_permitted_parameters
     ActionController::Base.helpers.number_to_percentage(account.balance / objective.amount * 100, precision: 0)
   end
 
+  helper_method memoize def top_up_time
```

Comment:
It's a server time zone. It's still different from the client (browser) time zone.

Also `Time.now` doesn't take into account timezone. `Time.current` could do the job. But even that's wrong because on the server we don't set time zone per user request. It's the same for all and it's in UTC.

So, the equal code should be in JS.

Reply:
https://www.tutorialspoint.com/how-to-detect-user-timezone-in-javascript

This way has its disadvantages (the user can change time zone and the system would better ask if it's ok to show time in their current time zone, but it's unnecessary complication at this point of time).